pkg/privilege/privileges/ldap/ldap_common.go

```diff
@@ -52,49 +65,20 @@ type ldapAuthImpl struct {
 	ldapConnectionPool *pools.ResourcePool
 }
 
-func (impl *ldapAuthImpl) searchUser(userName string) (dn string, err error) {
-	var l *ldap.Conn
-
-	l, err = impl.getConnection()
-	if err != nil {
-		return "", err
-	}
-	defer impl.putConnection(l)
-
-	err = l.Bind(impl.bindRootDN, impl.bindRootPWD)
-	if err != nil {
-		return "", errors.Wrap(err, "bind root dn to search user")
-	}
-
-	result, err := l.Search(&ldap.SearchRequest{
-		BaseDN: impl.bindBaseDN,
-		Scope:  ldap.ScopeWholeSubtree,
-		Filter: fmt.Sprintf("(%s=%s)", impl.searchAttr, userName),
-	})
-	if err != nil {
-		return
-	}
-
-	if len(result.Entries) == 0 {
-		return "", errors.New("LDAP user not found")
-	}
-
-	dn = result.Entries[0].DN
-	return
-}
+func (impl *ldapAuthImplBuilder) build() *ldapAuthImpl {
```

Comment:
```suggestion
func (builder *ldapAuthImplBuilder) build() *ldapAuthImpl {
```
Renaming it to `builder`?

Reply:
Fixed. :+1: